### PR TITLE
101460 - Unable to save IPO-filter with no project

### PR DESCRIPTION
**Pull request by chormaz**
Made projectId nullable in db and implemented necessary changes in api to allow users to save ipo-filters without a specific project. 

[AB#101460](https://statoildeveloper.visualstudio.com/b6d97460-2e53-47ae-aad4-b41396d90828/_workitems/edit/101460)

### CODE REVIEW

File: src/Equinor.ProCoSys.IPO.Command/PersonCommands/CreateSavedFilter/CreateSavedFilterCommandHandler.cs [hunk 1]
```diff
@@ -67,6 +71,10 @@ public async Task<Result<int>> Handle(CreateSavedFilterCommand request, Cancella
 
         public async Task<string> GetProjectFromMainAsync(string projectName)
         {
+            if (projectName == null)
```
Comment (janid1967) [inline src/Equinor.ProCoSys.IPO.Command/PersonCommands/CreateSavedFilter/CreateSavedFilterCommandHandler.cs:74]: Undo! No need to check this here. You've already checked it outside method

**Comment by janid1967** [comment]
[Equinor.ProCoSys.IPO.Command.Tests/Validators/SavedFilterValidatorTests.cs](https://github.com/equinor/procosys-invitation-for-punch-out-api/pull/242/files#diff-bfb24e48c0cbdd8e528accfb6c6881d276c39beb1166995f9028cc05ac901228) need 2 more tests. Both tests with null as projectName, first test with Assert.IsFalse(result) second with Assert.IsTrue(result);

File: src/Equinor.ProCoSys.IPO.Domain/AggregateModels/PersonAggregate/Person.cs [hunk 2]
```diff
@@ -72,7 +72,7 @@ public void RemoveSavedFilter(SavedFilter savedFilter)
         }
 
         public SavedFilter GetDefaultFilter(Project project) =>
-            _savedFilters.SingleOrDefault(s => s.ProjectId == project.Id && s.DefaultFilter);
+            _savedFilters.SingleOrDefault(s => s.ProjectId == project?.Id && s.DefaultFilter);
```
Comment (janid1967) [inline src/Equinor.ProCoSys.IPO.Domain/AggregateModels/PersonAggregate/Person.cs:75]: Add unittest in PersonTests where project is null

File: src/tests/Equinor.ProCoSys.IPO.Domain.Tests/AggregateModels/PersonAggregate/SavedFilterTests.cs [hunk 3]
```diff
@@ -29,21 +29,13 @@ public void Setup()
         public void Constructor_SetsProperties()
```
Comment (janid1967) [inline src/tests/Equinor.ProCoSys.IPO.Domain.Tests/AggregateModels/PersonAggregate/SavedFilterTests.cs:29]: Add a second Constructor_SetsProperties() test where you test with null for Project and Assert.IsFalse(dut.ProjectId.HasValue)

File: src/tests/Equinor.ProCoSys.IPO.Domain.Tests/AggregateModels/PersonAggregate/SavedFilterTests.cs [hunk 4]
```diff
@@ -29,21 +29,13 @@ public void Setup()
...
         {
             Assert.AreEqual(TestPlant, _dut.Plant);
-            Assert.AreEqual(ProjectId, _dut.ProjectId);
+            Assert.AreEqual(ProjectId, _dut.ProjectId.Value);
             Assert.AreEqual(Title, _dut.Title);
             Assert.AreEqual(Criteria, _dut.Criteria);
             Assert.AreEqual(DefaultFilterValue, _dut.DefaultFilter);
         }
```
Comment (janid1967) [inline src/tests/Equinor.ProCoSys.IPO.Domain.Tests/AggregateModels/PersonAggregate/SavedFilterTests.cs:38]: remove extra blank lines

File: src/Equinor.ProCoSys.IPO.Query/GetSavedFiltersInProject/GetSavedFiltersInProjectQueryHandler.cs [hunk 5]
```diff
@@ -33,6 +33,20 @@ public class GetSavedFiltersInProjectQueryHandler: IRequestHandler<GetSavedFilte
                 where p.Oid == currentUserOid 
                 select p).SingleAsync(cancellationToken);
 
+            if(request.ProjectName == null)
```
Comment (janid1967) [inline src/Equinor.ProCoSys.IPO.Query/GetSavedFiltersInProject/GetSavedFiltersInProjectQueryHandler.cs:36]: Add unittest testing when request.ProjectName == null

**Review by janid1967 — CHANGES_REQUESTED** [review]
As commented

File: src/Equinor.ProCoSys.IPO.Domain/AggregateModels/PersonAggregate/Person.cs (hunk 2 above)
Comment (chormaz) [reply src/Equinor.ProCoSys.IPO.Domain/AggregateModels/PersonAggregate/Person.cs:75]: Done in CreateSavedFilterWithoutProject_ShouldSaveFilter()

File: src/tests/Equinor.ProCoSys.IPO.Domain.Tests/AggregateModels/PersonAggregate/SavedFilterTests.cs (hunk 4 above)
Comment (chormaz) [reply src/tests/Equinor.ProCoSys.IPO.Domain.Tests/AggregateModels/PersonAggregate/SavedFilterTests.cs]: Didn't see any - but file changed anyway due to new tests

File: src/tests/Equinor.ProCoSys.IPO.Domain.Tests/AggregateModels/PersonAggregate/SavedFilterTests.cs (hunk 3 above)
Comment (chormaz) [reply src/tests/Equinor.ProCoSys.IPO.Domain.Tests/AggregateModels/PersonAggregate/SavedFilterTests.cs:31]: Done

File: src/Equinor.ProCoSys.IPO.Command/PersonCommands/CreateSavedFilter/CreateSavedFilterCommandHandler.cs (hunk 1 above)
Comment (chormaz) [reply src/Equinor.ProCoSys.IPO.Command/PersonCommands/CreateSavedFilter/CreateSavedFilterCommandHandler.cs]: Removed

**Review by janid1967 — APPROVED** [review]

**Comment by janid1967** [comment]
I'll add some integration tests
